manifests/providers/params.pp

```diff
@@ -1,26 +1,48 @@
 # foreman::providers default parameters
 class foreman::providers::params {
   # Dependency packages for different providers supplied in this module
-  $apipie_bindings = true
+  $oauth = true
+  $json = (versioncmp($::rubyversion, '1.9') < 0)
+  $apipie_bindings = false
   $foreman_api = false
 
   # OS specific package names
   case $::osfamily {
     'RedHat': {
+      if versioncmp($::puppetversion, '4.0') >= 0 {
+        $oauth_package = 'puppet-agent-oauth'
```

Comment:
Please note that these AIO related package names depend on https://github.com/theforeman/foreman-packaging/pull/1095 and a forthcoming PR for the Debian package.